File: go/api/Sntp.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Sntp 对应 SNtp.py。
type Sntp struct {
	*session.ApiGroup
}

// NewSntp 创建 Sntp API 分组。
func NewSntp(s *session.Session) *Sntp {
	return &Sntp{ApiGroup: session.NewApiGroup(s)}
}

// GetSettings 对应 sntp/settings。
func (s *Sntp) GetSettings() (map[string]interface{}, error) {
	return s.S.Get("sntp/settings", nil, "api")
}

// Sntpswitch 对应 sntp/sntpswitch。
func (s *Sntp) Sntpswitch() (map[string]interface{}, error) {
	return s.S.Get("sntp/sntpswitch", nil, "api")
}

// Serverinfo 对应 sntp/serverinfo。
func (s *Sntp) Serverinfo() (map[string]interface{}, error) {
	return s.S.Get("sntp/serverinfo", nil, "api")
}

// Timeinfo 对应 sntp/timeinfo。
func (s *Sntp) Timeinfo() (map[string]interface{}, error) {
	return s.S.Get("sntp/timeinfo", nil, "api")
}

File: src/api/s_ntp.rs
//! SNtp API group (`api/SNtp.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// SNtp API group.
pub struct SNtp<'a> {
    conn: &'a Connection,
}

impl<'a> SNtp<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        SNtp { conn }
    }

    /// `sntp/settings`. SNTP settings.
    pub fn get_settings(&self) -> Result<Value> {
        get_value(self.conn, "api/sntp/settings")
    }

    /// `sntp/sntpswitch`. SNTP switch.
    pub fn sntpswitch(&self) -> Result<Value> {
        get_value(self.conn, "api/sntp/sntpswitch")
    }

    /// `sntp/serverinfo`. SNTP server information.
    pub fn serverinfo(&self) -> Result<Value> {
        get_value(self.conn, "api/sntp/serverinfo")
    }

    /// `sntp/timeinfo`. SNTP time information.
    pub fn timeinfo(&self) -> Result<Value> {
        get_value(self.conn, "api/sntp/timeinfo")
    }
}
